src/Avalonia.Animation/Animatable.cs

```diff
@@ -58,33 +60,55 @@ public Transitions Transitions
                 if (value is null)
                     return;
 
-                if (_previousTransitions is null)
-                    _previousTransitions = new Dictionary<AvaloniaProperty, IDisposable>();
+                if (_activeIterations is null)
+                    _activeIterations = new Dictionary<AvaloniaProperty, IDisposable>();
 
                 SetAndRaise(TransitionsProperty, ref _transitions, value);
             }
         }
 
+        internal protected void EnableTransitions() => _disableCount--;
+        internal protected void DisableTransitions()
+        {
+            _disableCount++;
+
+            if (_disableCount > 0 && (_activeIterations?.Count ?? 0) > 0)
+            {
+                foreach (var iterationKP in _activeIterations.ToList())
```

Comment:
@kekekeks @MarchingCube I really don't like duplicating this Dictionary for enumeration, do you guys have any non-alloc alternatives for this?

Reply:
Also, @MarchingCube i think im seeing a double-dispose issue again, im not sure where it's originating

Reply:
```cs
var iterations = _activeIterations;
_activaIterations = null;
foreach(var iterations ...
```

Reply:
Also, move dictionary creation to `PropertyChanged` and check for `!(_transitions?.Count>0)` instead of `_activeIterations is null `

Reply:
@kekekeks Okay thanks

Reply:
So for some reason it doesnt do double disposal anymore, seems to be getter/setter setting _activeIterations was the culprit.